src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -143,36 +144,93 @@ private void ImportCalli(int token)
+                    _interpreter.SetReturnValue((byte)integer);
+                    break;
+                case TypeFlags.Int16:
+                    Debug.Assert(integer >= short.MinValue && integer <= short.MaxValue);
+                    _interpreter.SetReturnValue((short)integer);
+                    break;
+                case TypeFlags.UInt16:
+                    Debug.Assert(integer >= ushort.MinValue && integer <= ushort.MaxValue);
+                    _interpreter.SetReturnValue((ushort)integer);
+                    break;
+                case TypeFlags.Int32:
+                case TypeFlags.UInt32:
+                    _interpreter.SetReturnValue(integer);
+                    break;
+                case TypeFlags.Int64:
+                case TypeFlags.UInt64:
+                    _interpreter.SetReturnValue(stackItem.AsInt64());
+                    break;
+                case TypeFlags.IntPtr:
+                case TypeFlags.UIntPtr:
+                    _interpreter.SetReturnValue(stackItem.AsIntPtr());
+                    break;
+                case TypeFlags.Single:
+                    _interpreter.SetReturnValue(stackItem.AsFloat());
                     break;
-                case StackValueKind.Int64:
-                    _interpreter.SetReturnValue(((Int64StackItem)stackItem).Value);
+                case TypeFlags.Double:
+                    _interpreter.SetReturnValue(stackItem.AsDouble());
                     break;
-                case StackValueKind.Unknown:
-                case StackValueKind.NativeInt:
-                case StackValueKind.Float:
-                    if (stackItem.Type == WellKnownType.Single)
-                        _interpreter.SetReturnValue(((FloatStackItem)stackItem).Value);
-                    else if (stackItem.Type == WellKnownType.Double)
-                        _interpreter.SetReturnValue(((DoubleStackItem)stackItem).Value);
+                case TypeFlags.ValueType:
+                    _interpreter.SetReturnValue(stackItem.AsValueType());
                     break;
-                case StackValueKind.ByRef:
-                case StackValueKind.ObjRef:
-                    _interpreter.SetReturnValue(((ObjectRefStackItem)stackItem).Value);
+                case TypeFlags.Interface:
+                case TypeFlags.Class:
+                    _interpreter.SetReturnValue(stackItem.AsObjectRef());
                     break;
-                case StackValueKind.ValueType:
+                case TypeFlags.CategoryMask:
+                case TypeFlags.Unknown:
+                case TypeFlags.Void:
+                case TypeFlags.Enum:
+                case TypeFlags.Nullable:
+                case TypeFlags.Array:
+                case TypeFlags.SzArray:
+                case TypeFlags.ByRef:
+                case TypeFlags.Pointer:
+                case TypeFlags.FunctionPointer:
+                case TypeFlags.GenericParameter:
```

Comment:
Everything after `GenericParameter` should be deleted from the switch. It is not part of the Category.